From 4adb53818ecc331cd4740e9adffb6ded3b2cbfdd Mon Sep 17 00:00:00 2001
From: Lukas Zeller <luz@plan44.ch>
Date: Fri, 17 Oct 2025 11:17:26 +0200
Subject: [PATCH] drivers/nvmem/rockchip-otp.c: add support for rv1103b

---
 drivers/nvmem/rockchip-otp.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/drivers/nvmem/rockchip-otp.c b/drivers/nvmem/rockchip-otp.c
index 001ceee56..bf4d5404e 100644
--- a/drivers/nvmem/rockchip-otp.c
+++ b/drivers/nvmem/rockchip-otp.c
@@ -822,6 +822,17 @@ static const struct rockchip_data rk3588_data = {
 	.reg_read = rk3588_otp_read,
 };
 
+static const char * const rv1103b_otp_clocks[] = {
+	"usr", "sbpi", "apb", "phy",
+};
+
+static const struct rockchip_data rv1103b_data = {
+	.size = 0x80,
+	.clocks = rv1103b_otp_clocks,
+	.num_clks = ARRAY_SIZE(rv1103b_otp_clocks),
+	.reg_read = rk3568_otp_read,
+};
+
 static const char * const rv1106_otp_clocks[] = {
 	"usr", "sbpi", "apb", "phy",
 };
@@ -903,6 +914,12 @@ static const struct of_device_id rockchip_otp_match[] = {
 		.data = (void *)&rk3588_data,
 	},
 #endif
+#ifdef CONFIG_CPU_RV1103B
+	{
+		.compatible = "rockchip,rv1103b-otp",
+		.data = (void *)&rv1103b_data,
+	},
+#endif
 #ifdef CONFIG_CPU_RV1106
 	{
 		.compatible = "rockchip,rv1106-otp",
-- 
2.53.0

